enclave_manager/avalon_enclave_manager/work_order_processor_manager.py

```diff
@@ -0,0 +1,336 @@
+                    processed
+        Returns :
+            wo_resp - A JSON response of the executed work order
+        """
+        try:
+            self._kv_helper.set("wo-processing", wo_id,
+                                WorkOrderStatus.PROCESSING.name)
+
+            # Get JSON workorder request corresponding to wo_id
+            wo_json_req = self._kv_helper.get("wo-requests", wo_id)
+            if wo_json_req is None:
+                logger.error("Received empty work order corresponding " +
+                             "to id %s from wo-requests table", wo_id)
+                self._kv_helper.remove("wo-processing", wo_id)
+                return None
+
+        except Exception as e:
+            logger.error("Problem while reading the work order %s"
+                         "from wo-requests table", wo_id)
+            self._kv_helper.remove("wo-processing", wo_id)
+            return None
+
+        logger.info("Create workorder entry %s in wo-processing table",
+                    wo_id)
+        self._kv_helper.set("wo-processing", wo_id,
+                            WorkOrderStatus.PROCESSING.name)
+
+        logger.info("Delete workorder entry %s from wo-scheduled table",
+                    wo_id)
+        self._kv_helper.remove("wo-scheduled", wo_id)
+
+        logger.info("Validating JSON workorder request %s", wo_id)
+        if not self._validate_request(wo_id, wo_json_req):
+            return None
+
+        # Execute work order request
+
+        logger.info("Execute workorder with id %s", wo_id)
+        wo_json_resp = self._execute_work_order(wo_json_req)
+        wo_resp = json.loads(wo_json_resp)
+
+        logger.info("Update workorder receipt for workorder %s", wo_id)
+        self._wo_kv_delegate.update_receipt(wo_id, wo_resp)
+
+        if "Response" in wo_resp and \
+                wo_resp["Response"]["Status"] == WorkOrderStatus.FAILED:
+            logger.error("error in Response")
+            self._kv_helper.set("wo-processed", wo_id,
+                                WorkOrderStatus.FAILED.name)
+            self._kv_helper.set("wo-responses", wo_id, wo_json_resp)
+            self._kv_helper.remove("wo-processing", wo_id)
+            return None
+
+        logger.info("Mark workorder status for workorder id %s " +
+                    "as Completed in wo-processed", wo_id)
+        self._kv_helper.set("wo-processed", wo_id,
+                            WorkOrderStatus.SUCCESS.name)
+
+        logger.info("Create entry in wo-responses table for workorder %s",
+                    wo_id)
+        self._kv_helper.set("wo-responses", wo_id, wo_json_resp)
+
+        logger.info("Delete workorder entry %s from wo-processing table",
+                    wo_id)
+        self._kv_helper.remove("wo-processing", wo_id)
+        return wo_resp
+
+    # -------------------------------------------------------------------------
+
+    @abstractmethod
+    def _execute_work_order(self, input_json_str):
```

Comment:
Can we have a default implementation for `execute_work_order(config, work_order, ext_data="")`  same as singleton. For WPE, override `execute_work_order()`, call preprocess and then call  super class `execute_work_order()` function.